shared/api/image.go

```diff
@@ -34,6 +34,9 @@ type ImagesPostSource struct {
 	// For type "image"
 	Fingerprint string `json:"fingerprint" yaml:"fingerprint"`
 	Secret      string `json:"secret" yaml:"secret"`
+
+	// API extension: image_types
+	ImageType string `json:"image_type" yaml:"image_type"`
```

Comment:
@stgraber is it worth making ImageType the field name on all of the image related structs, so they are consistent with each other, as Type is taken already in this stuct. (I just fixed a bug in my code where I got confused between InstanceType and Type).

Reply:
api.Image.ImageType feels weird. To be fair this one ^ feels weird too but yeah, didn't have much of a choice as I can't break backward compatibility...

So effectively treating `Type` in this context as `SourceType` and using `ImageType` as `Type`.

Reply:
@stgraber yes the stuttering isn't great either.